src/lib/all_type_variant.hpp

```diff
@@ -49,13 +49,13 @@ namespace detail {
 #define DATA_TYPE_ENUM_VALUES BOOST_PP_SEQ_TRANSFORM(GET_ELEM, 1, DATA_TYPE_INFO)
 #define DATA_TYPE_STRINGS BOOST_PP_SEQ_TRANSFORM(GET_ELEM, 2, DATA_TYPE_INFO)
 
-// We use a boolean data type in the JitOperatorWrapper.
+// We use a boolean and value id data type in the JitOperatorWrapper.
 // However, adding it to DATA_TYPE_INFO would trigger many unnecessary template instantiations for all other operators
 // and should thus be avoided for compilation performance reasons.
-// We thus only add "Bool" to the DataType enum and define JIT_DATA_TYPE_INFO (with a boolean data type) in
-// "lib/operators/jit_operator/jit_types.hpp".
+// We thus only add "Bool" and "ValueID" to the DataType enum and define JIT_DATA_TYPE_INFO (with a boolean and value id
+// data type) in "lib/operators/jit_operator/jit_types.hpp".
 // We need to append to the end of the enum to not break the matching of indices between DataType and AllTypeVariant.
-enum class DataType : uint8_t { Null, BOOST_PP_SEQ_ENUM(DATA_TYPE_ENUM_VALUES), Bool };
+enum class DataType : uint8_t { Null, BOOST_PP_SEQ_ENUM(DATA_TYPE_ENUM_VALUES), Bool, ValueID };
```

Comment:
I just realized that this will have an impact on the compile time of pretty much everything. Any chance we could get around this?

Reply:
Additionally, `ValueID` is really odd in a DataType enum. Given how much of a core concept `DataType` is in hyrise, I think (i.e., am dead set) we cannot affort adding `ValueID` to it.

Reply:
I'd prefer that. That should include bool as well, correct?
As this does not impact compile times, maybe this would be better in a follow-up PR. Could you open an issue for this?

Reply:
Yes, the bool would then also not be needed. I will create an issue.

Reply:
I found an alternative solution which does not require the `ValueID` data type and requires a little bit less code.